### PR TITLE
Start adding some unit tests.

### PR DESCRIPTION
**Pull request by brendandburns**

### CODE REVIEW

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import unittest
+import mock
+
+from azure.cli.core._util import CLIError
+from azure.cli.command_modules.acs.custom import (
+    _validate_service_principal,
+    _build_service_principal
+)
+
+class spClient:
```
Comment (troydai) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py]: It is recommended by PEP8 to surround top level class with two blank lines.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        spMock = mock.Mock()
+        spMock.list.return_value = []
+        client = spClient(spMock, None)
+        _validate_service_principal(client, '27497b5e-7ea6-4ff2-a883-b3db4e08d937')
+
+        self.assertTrue(spMock.list.called)
+        spMock.list.assert_has_calls([mock.call(filter="servicePrincipalNames/any(c:c eq '27497b5e-7ea6-4ff2-a883-b3db4e08d937')")])
+
+
+    def test_validate_service_principal_fail(self):
```
Comment (troydai) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py]: One blank line is enough for class methods.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        spMock = mock.Mock()
+        spMock.list.side_effect=KeyError('foo')
+        client = spClient(spMock, None)
+        try:
+            _validate_service_principal(client, '27497b5e-7ea6-4ff2-a883-b3db4e08d937')
```
Comment (troydai) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py]: Using [assertRaise](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises) is a better approach to test expected exception here.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py (hunk 2 above)
Comment (tjprescott) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py]: @troydai can we make pylint enforce PEP8 standards?

Comment (troydai) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py]: Should be able to so, and I'll look into it. Some rules of PEP8 are not desirable, we need to choose a sub set of it.

Comment (brendandburns) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py]: done.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py (hunk 1 above)
Comment (brendandburns) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py]: done.

**Comment by brendandburns** [comment]
Comments addressed, please re-check.

Thanks!

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+class spClient:
+    def __init__(self, spMock, appMock):
+        self.service_principals=spMock
+        self.applications=appMock
+
+class AcsServicePrincipalTest(unittest.TestCase):
+    def test_validate_service_principal_ok(self):
+        spMock = mock.Mock()
+        spMock.list.return_value = []
+        client = spClient(spMock, None)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py]: I believe you don't need the `spClient`, just init an magicMock`, and assign the fields, like `service_principals`, with another magic mock object.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import mock
+
+from azure.cli.core._util import CLIError
+from azure.cli.command_modules.acs.custom import (
+    _validate_service_principal,
+    _build_service_principal
+)
+
+
+class spClient:
```
Comment (brendandburns) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py]: Removed.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py (hunk 4 above)
Comment (brendandburns) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py]: done.

**Comment by brendandburns** [comment]
Comments addressed, please re-check.

Thanks!

**Pull request merged**